openassessment/xblock/static/js/src/lms/components/WaitingStepDetailsPage.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Alert, Container, Row, Spinner,
+} from '@edx/paragon';
+import WaitingStepList from './WaitingStepList';
+
+const WaitingStepPage = ({ waitingStepDataUrl }) => {
+  // Abort controller to clean up API calls on unmount
+  const abortController = new AbortController();
+  const signal = abortController.signal;
+
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [waitingStepDetails, setWaitingStepDetails] = useState([]);
+
+  const fetchWaitingStepDetails = async () => {
+    // Clear error and display loading component
+    setLoading(true);
+    setError(false);
+    try {
+      // Retrieve waiting step data and decode JSON
+      const response = await fetch(waitingStepDataUrl, { signal });
+      const json = await response.json();
+      // Set table data
+      setWaitingStepDetails(json);
+    } catch (requestError) {
+      setError(true);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchWaitingStepDetails();
+
+    // Abort all requests on unmount
+    return function cancel() {
+      abortController.abort();
+    };
+  }, []);
+
+  const waitingStepPageContent = () => {
+    const oraDescriptionText = gettext(
+      'The "{name}" problem is configured to require a minimum of {min_grades} '
+      + 'peer grades, and asks to review {min_graded} peers.',
+    ).replace(
+      '{name}', waitingStepDetails.display_name,
+    ).replace(
+      '{min_grades}', waitingStepDetails.must_be_graded_by,
+    ).replace(
+      '{min_graded}', waitingStepDetails.must_grade,
+    );
```

Comment:
@xitij2000 This bit is JS code and the API doesn't send the fully translated string to the frontend.
I'll keep the current implementation since the rest of the JS code here also relies on it.